src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -156,22 +156,29 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
 
             // Create a token that will fire if we are disposed or if we get paused.
             using var source = CancellationTokenSource.CreateLinkedTokenSource(uncancelledToken.Value, disposalToken);
+            try
+            {
+                // Now actually synchronize the workspace.
+                var cancellationToken = source.Token;
+                cancellationToken.ThrowIfCancellationRequested();
 
-            // Now actually synchronize the workspace.
-            var cancellationToken = source.Token;
-            cancellationToken.ThrowIfCancellationRequested();
-
-            var client = await RemoteHostClient.TryGetClientAsync(_workspace, cancellationToken).ConfigureAwait(false);
-            if (client == null)
-                return;
+                var client = await RemoteHostClient.TryGetClientAsync(_workspace, cancellationToken).ConfigureAwait(false);
+                if (client == null)
+                    return;
 
-            using (Logger.LogBlock(FunctionId.SolutionChecksumUpdater_SynchronizePrimaryWorkspace, cancellationToken))
+                using (Logger.LogBlock(FunctionId.SolutionChecksumUpdater_SynchronizePrimaryWorkspace, cancellationToken))
+                {
+                    var workspaceVersion = solution.WorkspaceVersion;
+                    await client.TryInvokeAsync<IRemoteAssetSynchronizationService>(
+                        solution,
+                        (service, solution, cancellationToken) => service.SynchronizePrimaryWorkspaceAsync(solution, workspaceVersion, cancellationToken),
+                        cancellationToken).ConfigureAwait(false);
+                }
+            }
+            catch (OperationCanceledException)
```

Comment:
Same comment as below

```suggestion
            catch (OperationCanceledException) when (!disposalToken.IsCancellationRequested)
```